### PR TITLE
MES-1421: Get journals from DynamoDB by staffNumber path parameter

### PR DESCRIPTION
**Pull request by johnmorleybjss**
# Description and relevant Jira numbers
**MES-1421**
Going to DynamoDB to find journals by `staffNumber`. Dealing with any uncaught errors/unfound journals.

Ignoring all extra info in the `JournalWrapper` type for now (e.g. `hash`, `lastRefreshedAt`). Still needs the data wrangling to ensure current dates (MES-1496), however, if we're happy with the fetching functionality, let's merge early.

Some parts might look _very_ verbose, but this is partly down to TS strictness.

Looking for initial feedback on the structure/style - I will fill in more tests once we reach a point where these are correct.

## Pull Request checklist

- [x] [WIP] tag removed from PR title
- [x] PR has an understandable description

## Git feature branch checklist

- [x] branch name comply with our branching strategy
- [x] git branch contains relevant JIRA ticket number
- [x] branch rebased against the latest develop

## Sign off process checklist

- [x] Code has been tested manually
- [x] Tested by QA
- [x] PO's approval
- [x] Reviewers selected in Github
- [x] PR link added to JIRA

### CODE REVIEW

**Comment by chrisnappin** [comment]
Implementation looks good to me, just highlights a few things we're missing like configuration and error logging (will be on the backlog). How about commenting our code, JSDoc tags, etc?

File: src/functions/getJournal/framework/aws/DynamoJournalRepository.ts [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+}
+
+function getJournalTableName(): string {
+  let tableName = process.env.JOURNAL_DDB_TABLE_NAME;
+  if (tableName === undefined || tableName.length === 0) {
+    logger.warn('No journal table name set, using the default');
+    tableName = 'journal';
+  }
+  return tableName;
+}
```
Comment (rossfauldsbjss) [inline src/functions/getJournal/framework/aws/DynamoJournalRepository.ts]: Would like to see a test covering 404 - for a staff number not being found.

Comment (johnmorleybjss) [reply src/functions/getJournal/framework/aws/DynamoJournalRepository.ts]: Thanks, added.

**Comment by ammarhaiderbjss** [comment]
Have we looked at any alternatives to the standard AWS dynamo document client? It can get very nasty very quickly especially when we start to use expressions.

**Comment by johnmorleybjss** [comment]
Maybe [Dynamite](https://github.com/Medium/dynamite) @ammarhaiderbjss? Is this something we want to address in this PR? Or refactor later?

**Comment by johnmorleybjss** [comment]
Going to merge for now and get this working on dev. Happy to work on migrating to a different DDB client if this is what we want.

**Pull request merged**